sim/target-agnostic.mk

```diff
@@ -102,8 +102,8 @@ VERILATOR_MAKEFLAGS ?= -j8 VM_PARALLEL_BUILDS=1
 verilator = $(GENERATED_DIR)/V$(DESIGN)
 verilator_debug = $(GENERATED_DIR)/V$(DESIGN)-debug
 
-$(verilator) $(verilator_debug): export CXXFLAGS := $(CXXFLAGS) $(common_cxx_flags) $(VERILATOR_CXXOPTS) -D RTLSIM
-$(verilator) $(verilator_debug): export LDFLAGS := $(LDFLAGS) $(common_ld_flags)
+$(verilator) $(verilator_debug): export CXXFLAGS += $(common_cxx_flags) $(VERILATOR_CXXOPTS) -D RTLSIM
+$(verilator) $(verilator_debug): export LDFLAGS += $(common_ld_flags)
```

Comment:
There are several of these target-specific variable definitions that I modified.  While it is great to use simple variable types these statements cause`--warn-undefined-variables` to complain because you could potentially be using an undefined variable.  Since the expansion here is not expensive, switch to using simple recursive += 

If you really still want/need these to be simply expanded then adding an `ifndef` block for each of them to initialize them to be explicitly empty will also quell the `--warn-undefined-variables` messages.